test/TestMatchError.py

```diff
@@ -0,0 +1,51 @@
+from ansiblelint.errors import MatchError
+from ansiblelint.rules import AnsibleLintRule
+from ansiblelint.rules.AlwaysRunRule import AlwaysRunRule
+from ansiblelint.rules.BecomeUserWithoutBecomeRule import BecomeUserWithoutBecomeRule
+
+import pytest
+
+
+@pytest.mark.parametrize(
+    ('a', 'b'), (
+        (MatchError("foo"), MatchError("foo")),
+        # line is ignored in comparisions, even if it may be different
+        (MatchError("a", line={}), MatchError("a", line={"a": "b"}))
+    ))
+def test_matcherror_compare(a, b):
+    """Check that MatchError instances with similar attrs are equivalent."""
+    assert a == b
```

Comment:
I prefer better names here
```suggestion
    ('left_match_error', 'right_match_error'),
    (
        (MatchError("foo"), MatchError("foo")),
        # line is ignored in comparisions, even if it may be different
        (MatchError("a", line={}), MatchError("a", line={"a": "b"})),
    ),
)
def test_matcherror_compare(left_match_error, right_match_error):
    """Check that MatchError instances with similar attrs are equivalent."""
    assert left_match_error == right_match_error
```